### PR TITLE
[4.0] Media Manager update

### PR DESCRIPTION
**Pull request by laoneo**
Since the last media manager merge #18145 we fixed some smaller stuff and added a new feature to download the media file.

Ping @wilsonge, please do not squash to honor the work of the whole team.

### CODE REVIEW

File: administrator/components/com_media/Adapter/AdapterInterface.php [hunk 1]
```diff
@@ -68,13 +68,28 @@ public function getFile($path = '/');
 	 */
 	public function getFiles($path = '/');
 
+	/**
+	 * Returns a resource to download the path.
```
Comment (brianteeman) [inline administrator/components/com_media/Adapter/AdapterInterface.php]: surely you are not downloading the path but returning a path to the download?

File: administrator/language/en-GB/en-GB.com_media.ini [hunk 2]
```diff
@@ -16,6 +16,7 @@ COM_MEDIA_DELETE_ERROR="Error deleting the item."
 COM_MEDIA_DELETE_SUCCESS="Item deleted."
 COM_MEDIA_EDIT="Media Edit"
 COM_MEDIA_ERROR="An error occurred."
+COM_MEDIA_ERROR_NO_PROVIDERS="No filesystem providers are found. Please enable at least one filesystem plugin."
```
Comment (brianteeman) [inline administrator/language/en-GB/en-GB.com_media.ini]: +COM_MEDIA_ERROR_NO_PROVIDERS="No filesystem providers have been found. Please enable at least one filesystem plugin."

**Comment by laoneo** [comment]
Fixed the language errors, thanks @brianteeman.

**Comment by wilsonge** [comment]
Thanks to all involved on the [Media Manager](https://github.com/orgs/joomla-projects/teams/media-manager-improvements) team!!

**Pull request merged**